### PR TITLE
net: http: server: improve CRiMe utilities

### PR DESCRIPTION
**Pull request by cfriedt**
Add a cmake utility that generates Compressed Resources in Memory (CRiMe) for all files under a given directory.

Just a draft at the moment. My TODO list:
- [ ] create a C macro for easily defining CRiMe resources in source code (can't recall if this already exists)
- [ ] the above would basically require a detail, resource type enumeration, a common, handler that can be re-used for all static resources, etc.
- [ ] convert generator to python instead of cmake, maybe?

### CODE REVIEW

File: tests/net/lib/http_server/crime/prj.conf [hunk 1]
```diff
@@ -39,4 +42,18 @@ CONFIG_MAIN_STACK_SIZE=2048
...
-CONFIG_UART_PIPE=n
\ No newline at end of file
+CONFIG_NO_OPTIMIZATIONS=y
+CONFIG_DEBUG=y
+CONFIG_DEBUG_INFO=y
+
+# overlay-e1000.conf
+CONFIG_NET_L2_ETHERNET=y
+CONFIG_NET_QEMU_ETHERNET=y
+CONFIG_PCIE=y
```
Comment (jukkar) [inline tests/net/lib/http_server/crime/prj.conf:52]: These e1000 settings should not be set for tests which typically use the dummy interface.

File: tests/net/lib/http_server/crime/src/main.c [hunk 2]
```diff
@@ -4,85 +4,157 @@
...
+		.port = port_str,
+	};
+}
+
+static const struct http_service_desc *http_service_get_by_port(uint16_t port)
+{
+	const struct http_service_desc *svc;
+
+	HTTP_SERVICE_FOREACH(svc)
+	{
```
Comment (jukkar) [inline tests/net/lib/http_server/crime/src/main.c:40]: { to prev line

File: tests/net/lib/http_server/crime/src/main.c [hunk 3]
```diff
@@ -4,85 +4,157 @@
...
+
+	return NULL;
+}
+
+static const struct http_service_desc *
+http_service_get_resource(const struct http_service_desc *scv, const char *resource)
+{
+	const struct http_resource_desc *res;
+	HTTP_SERVICE_FOREACH_RESOURCE(svc, res)
+	{
```
Comment (jukkar) [inline tests/net/lib/http_server/crime/src/main.c:54]: ditto

**Review by jukkar — COMMENTED** [review]
Some minor nits while doing cursory glance of the code.

File: tests/net/lib/http_server/crime/prj.conf (hunk 1 above)
Comment (jukkar) [reply tests/net/lib/http_server/crime/prj.conf:52]: Also we can use loopback interface which does not need us to create network interface in the test if we do not need to tweak the low protocol data when it is being sent/received. Anyway, there are many examples for these in the tests directory.

File: tests/net/lib/http_server/crime/src/main.c (hunk 3 above)
Comment (cfriedt) [reply tests/net/lib/http_server/crime/src/main.c:54]: I think this is just the vs code auto formatter, but agred

**Pull request closed without merging**